core/authorship/impl/block_builder_factory_impl.cpp

```diff
@@ -35,11 +35,14 @@ namespace kagome::authorship {
     primitives::BlockHeader header;
     header.number = number;
     header.parent_hash = parent_hash;
-    header.digests = std::move(inherent_digests);
-
-    OUTCOME_TRY(r_core_->initialise_block(header));
-    return std::make_unique<BlockBuilderImpl>(std::move(header),
-                                              r_block_builder_);
+    header.digest = std::move(inherent_digest);
+
+    if (auto res = r_core_->initialise_block(header); not res) {
```

Comment:
Perhaps it'd make sense to rename methods accordingly, to avoid inconsistency. I mean initialize, finalize.

Reply:
In a separate PR, not related here, just increases diff